src/main/java/com/openlattice/authorization/Ace.java

```diff
@@ -74,19 +74,24 @@ public OffsetDateTime getExpirationDate() {
     }
 
     @Override public boolean equals( Object o ) {
-        if ( this == o )
+        if ( this == o ) {
             return true;
-        if ( o == null || getClass() != o.getClass() )
+        }
+        if ( o == null || getClass() != o.getClass() ) {
             return false;
+        }
         Ace ace = (Ace) o;
-        return hashValue == ace.hashValue &&
+        return hashCode() == ace.hashCode() &&
                 Objects.equals( principal, ace.principal ) &&
                 Objects.equals( permissions, ace.permissions ) &&
                 Objects.equals( expirationDate, ace.expirationDate );
     }
 
     @Override public int hashCode() {
-        return Objects.hash( principal, permissions, expirationDate, hashValue );
+        if ( hashValue == 0 ) {
```

Comment:
This class returns the internal enum set, and while I didn't see any external modifiers in our codebase, if we are making hashCode immutable we should set it to return Collections.unmodifiableSet to enforce immutability now that hash code assumes no changes.

Reply:
👍🏻 makes sense, will follow up

Reply:
https://github.com/openlattice/api/pull/488

Reply:
https://github.com/openlattice/conductor-client/pull/993